### PR TITLE
fix(build): support make dev on native Windows

### LINKED ISSUE #1885: [Bug]: make dev fails on native Windows (doctor prereq + Unix-only backend dev target)
### Environment

- **Commit:** `24f2fba18` (v0.81.0-15)
- **OS:** Windows 11
- **Toolchain:** GNU Make 4.4 (ezwinports), Go 1.26 (mise), pnpm (mise), no `sh.exe` on PATH

### Summary

`make start-windows` (production) works. `make dev` does not on native Windows — it's blocked by **two independent points**, and there's a design inconsistency between how `dev` and `run`/`start` inject the pprof env var that makes `dev` the odd one out.

### 1. `Makefile:156` — the `doctor` prerequisite aborts the whole target

`dev: doctor` runs `@scripts/doctor` (`Makefile:153`), a bash script. With no `sh.exe` on PATH, Make hands it to cmd.exe:

```
process_begin: CreateProcess(NULL, env bash C:\...\scripts\doctor, ...) failed.
make (e=2): The system cannot find the file specified.
make: *** [Makefile:153: doctor] Error 2
```

`doctor` is explicitly optional — its own header says it exits 0 if `pre-commit` is missing "so callers (e.g. `make dev`) don't block on an optional tool". But the `doctor` target's recipe isn't error-tolerant, so it takes `dev` down with it on Windows.

### 2. `apps/backend/Makefile:208` — the `dev` target is Unix-only

```make
dev: build
	@KANDEV_DEBUG_PPROF_ENABLED=true exec ./$(BUILD_DIR)/$(BINARY_NAME)$(EXE) __backend
```

Unix env-var prefix + `exec`. cmd.exe reads the variable name as a program name:

```
'KANDEV_DEBUG_PPROF_ENABLED' is not recognized as an internal or external command
make: *** [Makefile:208: dev] Error 1
[kandev] backend exited (code=2, signal=null)
```

This fires **after** the full CGO build (~3 min), so the failure is costly. `run` (`:203`) and `start-debug` (`:213`) share the pattern. The backend Makefile header (`:20`) does document these as Unix-only — but that collides with point 3.

### 3. The real inconsistency: `dev` injects pprof via Make, while `run`/`start` do it via env

- `apps/cli/src/dev.ts:56` launches the backend with `make -C apps/backend dev`, wrapped in `withWinjobWrap` — the **Windows-specific** Ctrl-C Job-Object wrapper. So the dev launcher is explicitly built for Windows, yet routes through a target the Makefile calls Unix-only.
- `apps/cli/src/run.ts:204` and `apps/cli/src/start.ts:83` set `KANDEV_DEBUG_PPROF_ENABLED: "true"` in the **backend process env** (`extra`), and it's forwarded through the supervisor manifest (`apps/cli/src/supervisor/manifest.ts:32`). Fully cross-platform.
- `dev.ts` does **not** set it in env — its `baseExtra` only sets `KANDEV_DEBUG_DEV_MODE`. It's the only entry point still relying on the Makefile recipe to inject pprof.

So `run` and `start` already solved this the portable way; `dev` is the one left behind.

### Suggested directions

**Idiomatic (matches run/start):** move `KANDEV_DEBUG_PPROF_ENABLED: "true"` into `dev.ts`'s `baseExtra` and have the CLI invoke a plain, cross-platform run target instead of the pprof-injecting `dev`. The Makefile's `dev` target keeps its Unix-only status without contradicting the launcher.

**Minimal (Makefile-only):** give `dev`/`run`/`start-debug` a `ifeq ($(OS),Windows_NT)` branch using the existing `set VAR=VAL&` + `$(subst /,\,...)` pattern already present in this file (`CGO_PREFIX`, `:29`) and in the root `start-windows` target.

Either way, point 1 (`doctor`) needs the recipe made error-tolerant on Windows.

I have a working, locally-tested candidate patch and can open a PR if a direction is preferred.

### Repro

```
make dev
```
from PowerShell, native Windows, no `sh.exe` on PATH.

### PR DESCRIPTION
**Pull request by JnManso**
`make dev` never started on native Windows: it aborted first on the `doctor` prerequisite (a bash-only script run under cmd.exe) and, past that, on the backend `run`/`dev`/`start-debug` targets, which used Unix env-prefix + `exec` syntax cmd.exe reads as a program name — so the dev launcher that already wraps the backend in the Windows `winjob` Ctrl-C helper could never reach a running backend.

## Important Changes

- `Makefile` `doctor`: add a `Windows_NT` branch that prints a skip note and exits 0, honoring the script's own "optional, non-blocking" contract instead of taking `dev` down with it.
- `apps/backend/Makefile` `run`/`dev`/`start-debug`: add OS branches via new `DEV_ENV`/`DEBUG_ENV`/`EXEC`/`RUN_BIN` vars, matching the file's existing `CGO_PREFIX` pattern (Windows uses `set VAR=VAL&` and a backslashed path, no `exec`). The Unix branch expands to byte-identical commands, so Linux/macOS is unaffected.

## Validation

- `make dev` end-to-end on native Windows 11 (PowerShell / cmd.exe, no `sh.exe` on PATH): backend came up on `:38429` with `pprof endpoints registered` (confirming the env prefix reached the process), Vite web on `:37429`, browser opened; clean teardown, no orphan processes, port freed.
- `make -C apps/backend {run,dev,start-debug} --dry-run` confirmed the expanded recipe on Windows; verified the Unix branch expands to the same commands as before this change.
- `make doctor` exits 0 on Windows with the skip note.

## Possible Improvements

Low risk. A more idiomatic alternative would move `KANDEV_DEBUG_PPROF_ENABLED` into `dev.ts`'s env (as `run.ts`/`start.ts` already do) rather than the Makefile recipe; kept out of scope here. Unrelated pre-existing cosmetic noise remains from `apps/backend/Makefile`'s `$(shell command -v node ...)` under cmd.exe.

## Checklist

- [ ] I have performed a self-review of my code.
- [ ] I have manually tested my changes and they work as expected.
- [ ] My changes have tests that cover the new functionality and edge cases.
- [ ] If my change touches UI files (`apps/web/`), I have added or updated Playwright e2e tests in `apps/web/e2e/` and verified them with `make test-e2e`.
- [ ] I checked whether this affects public docs in `docs/public/**` and updated them or noted why no docs change is needed.

Closes #1885

<a href="https://cubic.dev/pr/kdlbs/kandev/pull/1886?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -150,7 +150,11 @@ bootstrap-e2e:
 
 .PHONY: doctor
 doctor:
+ifeq ($(OS),Windows_NT)
+	@echo pre-commit hooks skipped on native Windows - run scripts/doctor from Git Bash to enable
+else
 	@scripts/doctor
+endif
```
[the remark this answers is not in the document]
Comment (JnManso) [reply Makefile]: Good catch — fixed in 5860ca67f. The guard now keys on `$(OS)$(MSYSTEM)`, which equals `Windows_NT` only in native cmd/PowerShell (where MSYSTEM is empty). Under Git Bash/MSYS, MSYSTEM is set, so the concatenation differs and `scripts/doctor` runs as before. Verified: `make doctor` from Git Bash now runs the script, while cmd/PowerShell still skips with the note.

File: apps/backend/Makefile [hunk 2]
```diff
@@ -35,6 +36,12 @@ ifeq ($(OS),Windows_NT)
   GO_DARWIN_ARM64 = set CGO_ENABLED=0& set GOOS=darwin& set GOARCH=arm64&
   GO_DARWIN_AMD64 = set CGO_ENABLED=0& set GOOS=darwin& set GOARCH=amd64&
   EXE = .exe
+  # Run-target env prefixes and binary invocation. cmd.exe has no `NAME=VAL cmd`
+  # syntax (use `set NAME=VAL&`) and no `exec`; the binary path needs backslashes.
+  DEV_ENV = set KANDEV_DEBUG_PPROF_ENABLED=true&
+  DEBUG_ENV = set KANDEV_DEBUG_PPROF_ENABLED=true& set KANDEV_LOG_LEVEL=debug&
+  EXEC =
+  RUN_BIN = "$(subst /,\,$(BUILD_DIR)/$(BINARY_NAME).exe)"
```
[the remark this answers is not in the document]
Comment (JnManso) [reply apps/backend/Makefile]: Fixed in 5860ca67f — `RUN_BIN` now uses `$(EXE)` instead of a literal `.exe`. Same expansion in this branch (EXE = .exe here), but consistent with the rest of the Makefile.

**Comment by carlosflorencio** [comment]
There’s one portability issue I think we should fix before merging.

`apps/backend/Makefile` selects `cmd.exe` syntax whenever `OS=Windows_NT`, but Git Bash/MSYS also sets that value while running recipes through Bash. This makes commands such as:

```text
set KANDEV_DEBUG_PPROF_ENABLED=true& "bin\kandev.exe" __backend
```

run without exporting the environment variable.

Could we separate executable selection from shell-syntax selection and reuse the native-Windows guard from the root Makefile? Roughly:

```make
ifeq ($(OS),Windows_NT)
EXE := .exe
endif

ifeq ($(OS)$(MSYSTEM),Windows_NT)
# cmd.exe environment and execution syntax
else
# POSIX syntax, using ./bin/kandev$(EXE)
endif
```

It would also be useful to add a small `make -n` test covering Unix, native Windows (`MSYSTEM=`), and Git Bash (`MSYSTEM=MINGW64`). The current Windows CI runs the Go checks but doesn’t exercise these Make targets.

**Comment by carlosflorencio** [comment]
@codex review

**Comment by JnManso** [comment]
Done — pushed as `804d92c` (fix) and `e4eb121` (review follow-up).

**The portability fix**, split the way you sketched:
- `EXE` keyed on `$(OS)` alone, so any Windows build (native or Git Bash) keeps the `.exe` suffix.
- Shell syntax (env prefix, `exec`, path style — plus `CGO_PREFIX`/`RM`/`MKDIR` and the cross-compile prefixes, which shared the same latent bug) keyed on `$(OS)$(MSYSTEM)`, mirroring the root Makefile's `doctor` guard. Native cmd/PowerShell keeps the cmd branch; Git Bash/MSYS now takes the POSIX branch. Native Windows and Unix expansions are byte-identical to before.

**Verified end-to-end on Windows 11**, booting the backend both ways:
- Git Bash → `KANDEV_DEBUG_PPROF_ENABLED=true exec ./bin/kandev.exe __backend` → `pprof endpoints registered` (env var reaches the process). The old cmd-syntax form under bash gives **0** pprof lines — the bug, reproduced.
- PowerShell (no `sh.exe` on PATH, so make uses cmd.exe) → `set …&` form → boots, `pprof endpoints registered`.

**The `make -n` test** you suggested: `scripts/check-make-shells` asserts run/dev/start-debug across Unix / native Windows (`MSYSTEM=`) / Git Bash (`MSYSTEM=MINGW64`), wired into the Linux backend-tests job. It fails against the pre-fix Makefile (the three Git Bash cases) and passes after.

Follow-up in `e4eb121` (from the CodeRabbit/Copilot pass on the fix commit): the check harness is a bash script, so the target now prints a skip note on native cmd/PowerShell instead of choking on the shebang, and I corrected the comments that over-claimed portability. CI runs the real check on Linux.
